packages/nimble-components/src/table/specs/table-row-grouping-hld.md

```diff
@@ -0,0 +1,228 @@
+                :rowGroupState="${(x, c) => c.parent.getGroupRowState(x.index)}"
+                @on-row-group-expand-toggle="${(x, c) => c.parent.toggleRowExpanded(x.index)}"
+                >
+            </${DesignSystem.tagFor(TableGroupRow)}>
+        `)}
+        ${when((x, c) => !(c.parent as Table).getIsGroupRow(x.index), html<VirtualItem, Table>`
+            <${DesignSystem.tagFor(TableRow)}
+    ...
+```
+
+The one change necessary to support this properly is to change the virtualizer to use the current length of `getRowModel.rows()` for the `count` instead of the [current length of the table's data](https://github.com/ni/nimble/blob/f60dedf600147b16916f876b749a4e019c2a6308/packages/nimble-components/src/table/models/virtualizer.ts#L82) (see [prototype](https://github.com/ni/nimble/blob/5cd05ab733b00ebb33e44d53c7be30c980fa3398/packages/nimble-components/src/table/models/virtualizer.ts#L87)).
+
+### Column header grouping visual
+
+Ultimately, when a column has set itself to be grouped, the design calls for that to be represented with a visual in the header for that column. This will be handled in a similar fashion to how we represent sort direction, by providing a new `is-grouped` attribute on `TableHeader` to determine whether to show the appropriate icon.
+
+### Interfacing with TanStack
+
+As shown in the above example, the `TableGroupRow` will provide a button to control expand/collapse behavior. TanStack handles the modeling for which rows are expanded or collapsed (in addition to which rows are groups), and thus must be notified of an expand/collapse action. This will be accomplished through raising an event (`on-row-group-expand-toggle` in the example above). The `Table` will handle the event, and call `stopPropogation` on it, as there is not an immediate need to leak this into the public event API of the `Table`.
+
+Setting TanStack up to handle both row grouping and expanded state is similar to what is in place for sorting. One notable difference is that we will need to register a handler for TanStack's `onExpandedChange` handler, as we will be working through TanStack to update this state, and the `Table` will need a notification in order to re-render the rows. It would look similar to the following:
+
+Table index.ts
+
+```
+public constructor() {
+    super();
+    this.options = {
+        data: [],
+        onStateChange: (_: TanStackUpdater<TanStackTableState>) => {},
+        onExpandedChange: this.handleExpandedChange,
+        ...
+
+```
+
+## Alternative Implementations / Designs
+
+Consider upstreaming a change to TanStack to allow row grouping by more than one column value. While this may not affect the overall architectural design proposed here, it would likely alter it to support providing multiple values to render in the `TableGroupHeaderView`.
+
+## Open Issues
+
+1. Should we mark the table validity state as invalid if more than one column specify the same `group-index`?
```

Comment:
Is there a reason we wouldn't validate `group-index` the same way we validate `sort-index`? I'd think we should keep them consistent.

Reply:
re-opening jic it was missed since that change doesn't seem pushed yet